x-pack/test/functional/services/pipeline_editor.js

```diff
@@ -150,6 +153,37 @@ export function PipelineEditorProvider({ getService }) {
       }
     }
 
+    async assertPatterInputSyntaxHighlighting(expectedHighlights, filterClasses) {
+      const patternInputElement = await testSubjects.find(SUBJ_CONTAINER);
+      let highlightedElements = await patternInputElement.findAllByXpath('.//div[@class="ace_line"]/*');
+
+      highlightedElements = await Promise.all(highlightedElements.map(async (element) => {
+        const highlightClass = await element.getAttribute('class');
+        const highlightedContent = await element.getVisibleText();
+
+        return {
+          highlightClass,
+          highlightedContent
+        };
+      }));
+
+      if (filterClasses) {
+        highlightedElements = highlightedElements.filter(({ highlightClass }) =>
+          filterClasses.some(filterClass => filterClass !== highlightClass)
+        );
+      }
+
+      expect(highlightedElements.length).to.be(expectedHighlights.length);
+      await Promise.all(highlightedElements.map(async ({ highlightClass, highlightedContent }, index) => {
```

Comment:
Looking at the `map` function definition, I'm not seeing any async code in there. So I'm not sure you need the `async` and the `Promise.all`, do you?

Reply:
Also, you could probably use `highlightedElements.forEach` here instead of `highlightedElements.map` since you aren't actually mapping each item in `highlightedElements` to something else?